syntax = "proto2";

package fabupilot.common.monitor;

//import "modules/common/proto/header.proto";

message Meta {
  optional string git_commit = 1;
  optional string resource_commit = 2;
  optional string release_generation_time = 3;
  optional string release_type = 4;
}

message RecordItem {
  enum ErrorType {
    OBSTACLE_MISSEDINSPECTION = 101;
    OBSTACLE_FALSEDETECTION = 102;
    OBSTACLE_STATUS = 103;
    TRAFFICLIGHT_MISSEDINSPECTION = 201;
    TRAFFICLIGHT_FALSEDETECTION = 202;
    TRAFFICLIGHT_UNSTABLE = 203;
    EXPERIENCE_SWING = 301;
    EXPERIENCE_WHEEL = 302;
    EXPERIENCE_BRAKE = 303;
    DANGER_ACROSSLANE = 401;
    DANGER_COLLISION = 402;
    DANGER_CHOOSELANE = 403;
    STABILITY_SHUTDOWN = 501;
    STABILITY_MONITOR = 502;
    STABILITY_LOWFRAME = 503;
    STABILITY_HORN = 504;
    UNABLETOPASS_STOPLONGTIME = 601;
    UNABLETOPASS_UNABLETOPASS = 602;
    UNKNOWN = 701;
  }
  optional double timestamp_sec = 1;
  optional string text = 2;
  optional bool take_over = 3;
  optional double take_over_utm_x = 7;
  optional double take_over_utm_y = 8;
  optional string take_over_reason = 4;
  optional bool is_manual = 5;
  optional string time = 6;
  repeated string reason_path = 10;
  repeated string msgs = 11;
  optional int64 driver_id = 12;
  optional string driver_name = 13;
  optional string driver_license = 14;
  //add old field
  optional bool is_system_error_caused = 15;
}

message Record {
  enum Weather {
    SUNNY = 0;
    RAINY = 1;
    CLOUDY = 3;
    FOGGY = 4;
    SNOWY = 5;
  }

  enum RunPurpose {
    LOADED_OPERATION = 0;
    DEVELOPMENT_TEST = 1;
  }

  enum RoadCondition {
    ROAD_DRY = 0;
    ROAD_WET = 1;
    ROAD_SNOWY = 2;
  }

  enum DriverScene {
    CITY = 0;
    HIGHWAY = 1;
    PORT = 2;
  }

  optional string timestamp = 1;

  optional bool dirty = 2 [default = false];

  optional string vehicle = 3;

  optional string map = 4;

  optional string route = 5;

  optional string recorder = 6;

  optional RunPurpose run_purpose = 7;

  optional Weather weather = 8;

  optional RoadCondition road_condition = 11;

  optional DriverScene driver_scene = 12;

  repeated RecordItem record_list = 9;

  optional Meta meta = 10;
}
message BatchRecord {
    repeated Record records = 1;
}
